### ISSUE TITLE
Segfault in getBufferString

### ISSUE THREAD

**Issue by wbt729** [roboception/rc_genicam_api#14]
I'm getting a segfault when calling `Buffer::getTLType()`.

This happens in the `getBufferString()` function and can be fixed by the following code:
```
inline std::string getBufferString(const std::shared_ptr<const GenTLWrapper> &gentl,
                                   void *stream, void *buffer, GenTL::BUFFER_INFO_CMD cmd)
{
  std::string ret;

  GenTL::INFO_DATATYPE type;
  char tmp[1024]="";
  size_t size=sizeof(tmp);

  if (stream != 0 && buffer != 0)
  {
    //if (gentl->DSGetBufferInfo(stream, buffer, cmd, &type, &ret, &size) == GenTL::GC_ERR_SUCCESS)
    if (gentl->DSGetBufferInfo(stream, buffer, cmd, &type, &tmp, &size) == GenTL::GC_ERR_SUCCESS)
    {
      if (type == GenTL::INFO_DATATYPE_STRING)
      {
        ret=tmp;
      }
    }
  }

  return ret;
}
```

**Comment by heikohimu** [comment]
Thanks, I will add the fix.

**Comment by flixr** [comment]
Fixed with 38843f8fbf8eae676f4e94456a212c9c41a23745
